xen/arch/arm/mm.c

```diff
@@ -1287,20 +1305,31 @@ long arch_memory_op(int op, XEN_GUEST_HANDLE_PARAM(void) arg)
         if ( d == NULL )
             return -ESRCH;
 
-        for ( i = 0; i < req.num_frames; i++ )
+        if ( likely (req.num_frames <= 32) )
         {
-            xen_pfn_t pa, ma;
+            /* more than 95% cases for GSX operation */
+            xen_pfn_t pma[req.num_frames];
```

Comment:
although this compiles with our GCC it may not with older ones?

Reply:
It may not build with older one. But XEN itself would not be functional if built with ancient one.